source/transactions-convenient-api/tests/README.rst

```diff
@@ -0,0 +1,107 @@
+=====================================
+Convenient API for Transactions Tests
+=====================================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+The YAML and JSON files in this directory are platform-independent tests that
+drivers can use to prove their conformance to the Convenient API for
+Transactions spec.  They are designed with the intention of sharing some
+test-runner code with the CRUD, Command Monitoring, and Transaction spec tests.
+
+Several prose tests, which are not easily expressed in YAML, are also presented
+in this file. Those tests will need to be manually implemented by each driver.
+
+Server Fail Point
+=================
+
+See: `Server Fail Point <../../transactions/tests#server-fail-point>`_ in the
+Transactions spec test suite.
+
+Test Format
+===========
+
+Each YAML file has the following keys:
+
+- ``database_name`` and ``collection_name``: The database and collection to use
+  for testing.
+
+- ``data``: The data that should exist in the collection under test before each
+  test run.
+
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+
+  - ``description``: The name of the test.
+
+  - ``failPoint`` (optional): The ``configureFailPoint`` command document to run
+    to configure a fail point on the primary server.
+
+  - ``sessionOptions`` (optional): Names and values of options to pass to
+    ``MongoClient.startSession()``.
+
+  - ``operations``: Array of documents, each describing an operation to be
+    executed. Each document has the following fields:
+
+    - ``name``: The name of the operation on ``object``.
+
+    - ``object``: The name of the object on which to perform the operation. The
+      value will be either "collection" or "session0".
+
+    - ``arguments`` (optional): Names and values of arguments to pass to the
+      operation.
+
+    - ``error`` (optional): If ``true``, the test should expect the operation
```

Comment:
Let's replace this with `result` syntax from Transactions spec runner for consistency. This spec won't assert much about errors -- likely only that errors from `withTransactions` _don't_ have certain labels.

Additionally, we can add a test for the callback yielding a duplicate key error, which should propagate up through `withTransaction`.